src/test/ui/rfc1598-generic-associated-types/generic_associated_types_equals.rs

```diff
@@ -0,0 +1,18 @@
+// Copyright 2017 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+#![feature(generic_associated_types)]
+
+trait Foo {
+    type Bar<T=usize>;
+    type X<T> where T = f64;
```

Comment:
@nikomatsakis Are we going to support type defaults as part of this RFC?

Reply:
I would be inclined to have them work "as much as they do for regular associated types" (they are presently unstable there because we never could agree on the full semantics). It'd probably be more work to try and ban them.

Reply:
However, this is not the default syntax. Or maybe I didn't understand what you meant. The syntax might be more like:

```
type X<T> = f64;
```

Reply:
Maybe try something like this, which is basically the prime use-case:

```
trait Iterable {
    type Iter<'a>: Iterator;
}
```

Note that, in particular, I don't think we plan -- yet -- to support GATs with type parameters (we will eventually, we have to fixup some stuff in our trait system impl first). We should definitely parse them and so forth though.